### ISSUE TITLE
German Wikipedia (using ZIM chunks) articles are not all rendered properly with 2.4

### ISSUE THREAD

**Issue by uwemock** [kiwix/kiwix-android#794]
I am running Kiwix 2.2 Build 53 from the Google Play Store.

I have a couple of ZIM files on my SD card, such as german Wikibookx, german Wikiquote, german and english Wikivoyage... They all work fine. However, I could not get the german Wikipedia up and running.

I tried to use the -nopic and the-novid files, but they just won't show up in the Kiwix content list.

I tried to split the -novid into 2GB chunks, but it still doesn't show up in Kiwix. Yes, I checked the naming, and all chunks are in the same directory.

The german Wikipedia is also not in the remote files list, so I could not test if it works to have Kiwix download the files.

Any hints?

**Comment by Martin5372** [comment]
Hi,
I just have the same problem and found your entry :-)
I tried the following files:

MD5 hashes: created from my downloaded files:
f8af88cc286c0b4e8d311e5c89d9af5f *wikipedia_de_all_nopic_2018-05.zim
f7119ed39fcf12f34afa9968d4aa5fc4 *wikipedia_de_all_novid_2018-05.zim
a45623bd6993b3f7c374082e4c2b5408 *wikipedia_de_all_novid_2018-06.zim

The MD5 hash for wikipedia_de_all_novid_2018-06.zim.md5 is OK:
> http://download.kiwix.org/zim/wikipedia/wikipedia_de_all_novid_2018-06.zim.md5
> a45623bd6993b3f7c374082e4c2b5408  wikipedia_de_all_novid_2018-06.zim
For the others I found no reference hash file.

I want to use the latest wikipedia_de_all_novid_2018-06.zim on my Android smartphone (*.zimaa, ...).
It failed so I tried to open the ZIM-file in Kiwix for windows (V0.9).
But all of the above files reported the same error message:
Message in German: "* kann nicht geladen werden. Bist du sicher, dass dies eine ZIM-Datei ist?"
Message in English: "Unable to load *. Are you sure this is a ZIM file?"

A test with the following file
d57cc00bb3f10c057ed0889f4a39d70d *wikipedia_en_ray_charles_2015-06.zim
in Kiwix V0.9 for Windows was successfull!
I guess that the german ZIM files are corrupt.

**Comment by FTno** [comment]
Had the same problem with the english Wikipedia file 'wikipedia_en_all_novid_2018-05.zim'. Other zim files like Wikivoyage, Wikiquote and some more worked fine.

Then I updated to v2.3 (kiwix-2.3_arm64-v8a.apk - https://sourceforge.net/projects/kiwix/files/android/2.3/) and the Wikipedia file worked.

But the search feature does not work for the Wikipedia file, just a blank page whatever I search for.
Hope this will be fixed for the next version.

**Comment by Martin5372** [comment]
I just switched from Kiwix 2.2 (latest version in the Play Store) to 2.3 from 
https://sourceforge.net/projects/kiwix/files/android/2.3
thanks to FTno :+1: 

With this new Kiwix version for Android the described access problem for my ZIM file
a45623bd6993b3f7c374082e4c2b5408 *wikipedia_de_all_novid_2018-06.zim
was fixed.

But I have the same problem like FTno:
The search function is working, but for most of the clicked articles I get a blank page.
The blank page keeps active and the CPU utilization is low = no background activity and no progress.

**Comment by mhutti1** [comment]
Could you try the nighties here: http://download.kiwix.org/nightly/2018-06-08/ and see if this fixes it?

**Comment by mhutti1** [comment]
How about https://drive.google.com/open?id=1nI8vzMB-cXyrDLL9iPWXa7RCXekciCgo  ?

**Comment by Martin5372** [comment]
Hello mhutti1,
here are the results for my smartphone with Android 7.1.2 (LineageOS 14.1-20180628):

- Kiwix 2.4 + wikipedia_de_all_novid_2018-06.zim (ZIM splitted in 4000 MB(!) pieces):
This combination did not work. Still the same problems:
> The search function is working, but for most of the clicked articles I get a blank page.
> The blank page keeps active and the CPU utilization is low = no background activity and no progress.

Then I tried the following combination on my smartphone:
- Kiwix 2.4 + wikipedia_de_all_novid_2018-06.zim (ZIM splitted in 1500 MB(!) pieces):
Result: This combination works!! :-)
Very fast access to all articles and pictures and no blank screen. :-)

I already checked the SDCARD with H2testw: no errors.
Afterwards I make the same test again 4000 MB pieces vs. 1500 MB pieces.
The result is still the same.

Thank you very much for your help!
Please let me know, if you need more information to identify the size problem.

--------

Does your Kiwix version 2.4 requires a newer android version?
On my old tablet with android 4.2.2 this version is not stable:
black screen, App freezes, ... it seems, that the behaviour is independent from the ZIM files.
Is there a way to fix it?
The plan was to use Kiwix with wikipedia_de_all_novid_2018-06.zim on my old tablet, too.

have a nice day!

**Comment by uwemock** [comment]
While Kiwix 2.3 (from the above link) was able to read the german Wikipedia, the newer 2.4 that came through the Play Store can't read my files. It even refused to read files that were fine with 2.2. So I went back to 2.3.

Sorry, guys - this is a mess. Release versions should always be able to read the format of files that have been published to deliver the application workload.

**Comment by mhutti1** [comment]
@Martin5372 you should be able to go .zimaa -> .zimaz ->.zimba -> .zimbz -> .zimzz

**Comment by mhutti1** [comment]
Does this problem occur with 2000MB chucks (the size we use automatically for our downloads)?

**Comment by uwemock** [comment]
I tried with Kochwiki which is only 60MB large. Kiwix 2.2 and 2.3 will read the file with no problems, Kiwix 2.4 just quits after a few seconds of white screen.

**Comment by kelson42** [comment]
@mhutti1 What would be the best approach you would recommend to @uwemock to share with us his Kiwix Android crash log?

**Comment by Martin5372** [comment]
@mhutti1
> @Martin5372 you should be able to go .zimaa -> .zimaz ->.zimba -> .zimbz -> .zimzz

for wikipedia_en_all_novid_2018-05.zim I tried it with 1500 MB chunks with Kiwix 2.3 on my old tablet.
It's working :-)

> Does this problem occur with 2000MB chucks (the size we use automatically for our downloads)?

for wikipedia_en_all_novid_2018-05.zim I tested afterwards 2000 MB chunks with Kiwix 2.3 on my old tablet.
It's working :-)

thank you very much!

**Comment by kelson42** [comment]
@mhutti1 I can confirm that we have here something really buggy. Depending of the content and chunk size, Kiwix for Android seems not always able to deal properly with it. I have splitted `wikipedia_de_all_novid_2018-06` in 3500 MB chunks and then I can confirm that not all contents (images, pages) can be loaded properly... but if I try with kiwix-serve, with the same files, everything works fine.

**Comment by mhutti1** [comment]
@kelson42 hmm interesting. If standard sized chunks work then this isn't a major issue at the moment but we should certainly look into the root cause behind instability with larger chunk sizes. Possibly the use of Java int's overflowing somewhere I am not sure.

**Comment by Wikinaut** [comment]
I run wikipedia_de_all_novid_2018-06 but a search for "PDP" does not give any result (but pages like "PDP-11" are present, when I follow a link on the page "VAX" to it.

@kelson42 : pls. can you confirm, that this not-fully working search or index again breaks the Kiwix functionality (it's a show-stopper again, in my view).

**Comment by Wikinaut** [comment]
@Martin5372 can you help me, again, I found Kiwix not fully working. I downloaded and split wikipedia_de_novid_2018-06 to 2000MB chunks, like in the tutorial, but there are no results for a search for "PDP". Can you confirm that a split into smaller chunks really helps?

**Comment by mhutti1** [comment]
Hi @Wikinaut this is a known bug with our C++ backend. In larger files with a full text index the index must fit in a chunk for it to work unfortunately. We should be falling back to title search in this case, thanks for providing a specific reproduction case. I will look into this and see what is happening.

**Comment by Wikinaut** [comment]
@mhutti1 and @kelson42 Yesterday, I made a larger check, here are my findings, and I willing to help and to fix this issue very soon - I want to have the Kiwix running since years, but not a single German (or other version) worked for me in the last three years. This leaves me quite puzzled, because you are promoting Kiwix so much and loud, but as it never works here, I really doubt what is going on... well, let's try to fix this.)

* HUAWEI P8 LITE 2017
* Android 7.0
* SD CARD SAMSUNG PRO 128 GB
* KIWIX Android App 2.4 Build 1181980

I tried these files yesterday:

* Wikipedia_de_all_novid_2018_06 chunks of 1000 MB (almost no entries at all, only a very few)
* Wikipedia_de_all_novid_2018_06 chunks of 2000 MB (no entries for "pdp") 
* Wikipedia_de_all_novid_2018_06 chunks of 1000 MB (no entries for "pdp")

**Comment by IrgendwerA8** [comment]
I think this is a very serious issue too. On my Android 4.1.2 device, I'm not able to open a more current version of the German Wikipedia, even when spitted to the standard 2000MB chunk size:

Official 2.2: Fehler: Die ausgewählte Datei ist keine gültige ZIM-Datei.
2.3 from above: Crashes during start-up
2.4 from above: Starts and "seems" to open the ZIM after selecting, but after a second or so switches to black screen and hangs.

No problems at all with the two year old German Wikipedia and 2.2.

**Comment by kelson42** [comment]
@IrgendwerA8 What happens if you do not split it?

**Comment by IrgendwerA8** [comment]
I cannot put it on my FAT32 formatted SD-Card? ;)

**Comment by kelson42** [comment]
@IrgendwerA8 No way to get a card using exFAT? What you describe is a bug, but - up to now - to get full featured new ZIM, users will have to keep them in one (so, stop to split them)... so stop use FAT32 as a filesystem.

**Comment by uwemock** [comment]
Things are getting weird...
While Kiwix 2.3 is fully functional on my phone, the same version behaves differently on my tablet. I can see the start page of each ZIM, and I can also follow links to single articles. But I cannot use the search function. The results list remains empty.
This occurs on the -nopic Wikipedia split into 2000M chunks as well as on the 60MB Kochwiki.

**Comment by IrgendwerA8** [comment]
@kelson42 Just found my assumption confirmed by testing it: My device doesn't support exFAT and the microSD isn't recognized then. So splitting the data seems to be the only solution. I wonder what happened to the zim files that older ones work flawlessly and newer ones aren't valid anymore?

**Comment by uwemock** [comment]
Actually, the title holds for my original problem: The german Wikipedia did not work in Kiwix 2.2. This problem has been resolved by Kiwix 2.3.

Unfortunately, two things happened:
1) Kiwix 2.4 messed it all up again.
2) People having problems with splitting ZIM files up took over my thread.

I should, again, note that my problems with search not working in Kiwix 2.3 on my tablet apply to the Kochwiki ZIM file which is 60 MB in size (this is megabytes, not gigabytes) - and, of course, I didn't split this file up in smaller parts.

So I recommend that someone should file an issue for the splitting topic.

**Comment by mhutti1** [comment]
I have done some testing. My results are:

Full 30GB German file -> Searching "PDP" works (is a bit slow, be sure you don't mistake this for no results)
Split into ~2000MB chunks -> searching pd searches using the fallback searchsuggestions method. Searching pdp tries to do this but finds nothing. Searching pdp-11 gives nothing as well. If we however use developer tools to search PDP-11 and don't reset the capitalisation then it returns the singular correct result. @kelson42 this is definitively an issue with seachsuggestions. Likely a regression if this didn't happen before.
Haven't tried other sizes because I think I have found the issue.

**Comment by mhutti1** [comment]
Some more details. It isn't the length of a string that causes this issue. "hello" works fine. It might be the "-" or possibly the capitalisation.

**Comment by Wikinaut** [comment]
@mhutti1 V i e l e n   D a n k

**Comment by mhutti1** [comment]
@uwemock Could you try wiping the app data (not deleting the ZIM files) from settings.

**Comment by kelson42** [comment]
This bug (big files not open or all articles not rendered properly) has been fixed and is available in 2.4 (latest build) and in nightlies.

**Comment by IrgendwerA8** [comment]
@kelson42 Sorry to say, but no, on my device the local selection of zim files is still empty...

**Comment by kelson42** [comment]
@IrgendwerA8 This ticket has gathered many kind of different use cases. We think to have fixed the most important one, if you have still a problem, please open a new ticket and share again/copy your details after a new test with 2.4.

**Comment by Wikinaut** [comment]
@kelson42 @mhutti1 I installed (beta tester) the 

* Kiwix 2.4 Build 1182160 from Google Play Store and
* wikipedia_de_all_novid_2018-06 .zim with 2000 MB ( 1,95 GB) chunks

and I confirm that this **works** now.

**Comment by Wikinaut** [comment]
@kelson42 Did I say "Thank you" ? Thank you!

**Comment by uwemock** [comment]
I tried the current release from the Play Store (Kiwix 2.4 Build 1182160), and I deleted the app data after the upgrade.

It works on my Samsung S5 mini (Android 6).
It doesn't work on my Samsung Galaxy Tab A6, Android 5.

On the Galaxy Tab, I have the following issues:
* The UI is in English while the device is set to German.
* Article search works on smaller file, e.g. KochWiki, not on Wikipedia.
* Sometimes, accessing an article closes the app.
